lib/policial/client.rb

```diff
@@ -0,0 +1,41 @@
+module Policial
+  # Public: Starting with an Octokit client and a pull request,
+  # it checks all changes introduced looking for style guide violations. It
+  # also accuses all present violations through comments on all relevant lines
+  # in the pull request.
+  class Client
+    attr_accessor :violations
+    attr_reader :octokit, :pull_request
+
+    def initialize(octokit = nil)
+      @octokit = octokit || Octokit
+    end
+
+    def investigate
+      @violations ||= StyleChecker.new(pull_request).violations
+    end
+
+    def accuse
+      return if violations.nil?
+
+      commenter = Commenter.new(pull_request)
+
+      violations.each do |violation|
+        if accusation_policy.allowed_for?(violation)
+          commenter.comment_violation(violation)
+        end
+      end
+    end
+
+    def pull_request=(pull_request)
+      @pull_request = pull_request
+      @pull_request.octokit = @octokit
```

Comment:
Indeed, that's so much better. I've added `Detective#brief` in order to accomplish that (`#file` might cause confusion with `PullRequest#files`). Thanks!